torch/_inductor/lowering.py

```diff
@@ -1225,6 +1225,211 @@ def apply_constraint(arg, fx_arg):
+make_fallback(torch.ops.aten.special_ndtri)
+make_fallback(torch.ops.aten.special_scaled_modified_bessel_k0)
+make_fallback(torch.ops.aten.special_scaled_modified_bessel_k1)
+make_fallback(torch.ops.aten.special_spherical_bessel_j0)
+make_fallback(torch.ops.aten.special_xlog1py)
+make_fallback(torch.ops.aten.special_zeta)
+make_fallback(torch.ops.aten.take)
+make_fallback(torch.ops.aten.threshold)
+make_fallback(torch.ops.aten.trace)
+make_fallback(torch.ops.aten._trilinear)
+make_fallback(torch.ops.aten.unfold_copy)
+make_fallback(torch.ops.aten.unsafe_split)
+make_fallback(torch.ops.aten.vdot)
+make_fallback(torch.ops.aten.view_as_complex)
+make_fallback(torch.ops.aten.view_copy)
+make_fallback(torch.ops.aten.xlogy)
+make_fallback(torch.ops.aten._adaptive_avg_pool3d_backward)
+make_fallback(torch.ops.aten.adaptive_max_pool2d_backward)
+make_fallback(torch.ops.aten.adaptive_max_pool3d_backward)
+make_fallback(torch.ops.aten.avg_pool3d_backward)
+make_fallback(torch.ops.aten.binary_cross_entropy_backward)
+make_fallback(torch.ops.aten.bitwise_or_)
+make_fallback(torch.ops.aten._cdist_backward)
+make_fallback(torch.ops.aten.diagonal_backward)
+make_fallback(torch.ops.aten._embedding_bag_dense_backward)
+make_fallback(torch.ops.aten.fractional_max_pool2d_backward)
+make_fallback(torch.ops.aten.fractional_max_pool3d_backward)
+make_fallback(torch.ops.aten.hardshrink_backward)
+make_fallback(torch.ops.aten.huber_loss_backward)
+make_fallback(torch.ops.aten._linalg_check_errors)
+make_fallback(torch.ops.aten.log_sigmoid_backward)
+make_fallback(torch.ops.aten.max_pool3d_with_indices_backward)
+make_fallback(torch.ops.aten.multilabel_margin_loss_backward)
+make_fallback(torch.ops.aten.multi_margin_loss_backward)
+make_fallback(torch.ops.aten._pdist_backward)
+make_fallback(torch.ops.aten._prelu_kernel_backward)
+make_fallback(torch.ops.aten.reflection_pad1d_backward)
+make_fallback(torch.ops.aten.replication_pad1d_backward)
+make_fallback(torch.ops.aten.smooth_l1_loss_backward)
+make_fallback(torch.ops.aten.soft_margin_loss_backward)
+make_fallback(torch.ops.aten.softshrink_backward)
+make_fallback(torch.ops.aten.squeeze_copy)
+make_fallback(torch.ops.aten.linalg_pinv.atol_rtol_tensor)
+make_fallback(torch.ops.aten.segment_reduce.default)
+make_fallback(torch.ops.aten._segment_reduce_backward.default)
+make_fallback(aten.angle)
+make_fallback(aten.cholesky_inverse)
+make_fallback(aten.cholesky_solve)
+make_fallback(aten._fft_r2c)
+make_fallback(aten.histogram.bin_ct)
+make_fallback(aten._histogramdd_bin_edges.default)
+make_fallback(aten._histogramdd_from_bin_cts.default)
+make_fallback(aten.index_reduce)
+make_fallback(aten.masked_scatter)
+make_fallback(aten.to_sparse)
+make_fallback(aten.triangular_solve)
+make_fallback(aten.expand_copy)
+make_fallback(aten.zeros)
+
+make_fallback(torch.ops.prims.amin)
+make_fallback(torch.ops.prims.gcd.default)
+make_fallback(torch.ops.prims.shift_left.default)
```

Comment:
hm, I wonder how did these appear, we must be using some questionable refs

Reply:
I guess it's comming from unwanted `register_decomposition` for an inplace version

Reply:
We have lowerings for the aten versions of some of those.  Perhaps we shouldn't actually be falling back.

Reply:
Yeah we have or could create trivial pointwise lowerings for a lot of those, but we shouldn't be seeing prims in the graphs we are getting.

Reply:
Yes, meant to ask you guys about this. I've been looking into it and it's a couple of different things:

- For `shift_left`, I managed to get rid of the prim fallback, just had to add a missing implementation to `codegen/common.py`
- For `shift_right`, same story, but the naive implementation gives the wrong results on cuda somehow. Need to look into this some more.
- For `gcd` I think we just need to actually implement the prim in inductor, it's not there. Alternatively, we can drop the gcd decomposition from the inductor list and register a fallback for `aten.gcd` instead, probably better.
- For `amin`, the issue is that the reference for `linalg.norm` is calling `refs.amin` instead of `torch.amin`. If you fix that this issue no longer comes up, but then some mypy checks fail - need to sort that out.

Reply:
Ok, let's fix linalg.norm decomp to call torch.amin, mypy shouldn't be too hard, and even if it is, leave a comment to ignore it?
As for shift_right/left, yeah we can have lowering, I still wonder how did prim get into the graph.

Reply:
Should be all set.